src/parser/deserializer.ts

```diff
@@ -234,6 +305,7 @@ function deserializeObject(
       if (raw) {
         value = buffer.slice(index, index + objectSize);
       } else {
+        options.validation = { utf8: shouldValidateKey };
```

Comment:
in order to avoid mutating the options object, let's do something like:
```
let objectOptions = options;
if (!globalUTFValidation) { 
   objectOptions = { ...options, validation: { utf8: shouldValidateKey }  }
}
```
and then let's make a similar change to wrap the arrayOptions (no need to declare a new var though, just:
```
if (!globalUTFValidation) { 
   arrayOptions = { ...options, validation: { utf8: shouldValidateKey }  }
}
```